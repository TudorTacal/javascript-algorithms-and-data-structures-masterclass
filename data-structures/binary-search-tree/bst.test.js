const { BST } = require('./bst');

describe('BinarySearchTree', () => {
  test('#insert should insert a node when 0 nodes', () => {
    let bst = new BST();

    bst.insert(1);

    expect(bst).toMatchSnapshot();
  });

  test('#insert should insert a node when multiple nodes', () => {
    let bst = new BST();

    bst.insert(10);
    bst.insert(8);
    bst.insert(7);
    bst.insert(9);
    bst.insert(12);
    bst.insert(17);

    expect(bst).toMatchSnapshot();
  });

  test('#find should find a node when multiple nodes', () => {
    let bst = new BST();

    bst.insert(10);
    bst.insert(8);
    bst.insert(7);
    bst.insert(9);
    bst.insert(12);
    bst.insert(17);

    let result1 = bst.find(7);
    let result2 = bst.find(17);
    let result3 = bst.find(40);

    expect(result1).toEqual(true);
    expect(result2).toEqual(true);
    expect(result3).toEqual(false);
  });
});