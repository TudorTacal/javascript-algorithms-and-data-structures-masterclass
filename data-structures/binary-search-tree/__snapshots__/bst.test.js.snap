// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BinarySearchTree #bfs should traverse then tree 1`] = `
Array [
  Node {
    "left": Node {
      "left": Node {
        "left": null,
        "right": null,
        "value": 7,
      },
      "right": Node {
        "left": null,
        "right": null,
        "value": 9,
      },
      "value": 8,
    },
    "right": Node {
      "left": null,
      "right": Node {
        "left": Node {
          "left": null,
          "right": null,
          "value": 16,
        },
        "right": null,
        "value": 17,
      },
      "value": 12,
    },
    "value": 10,
  },
  Node {
    "left": Node {
      "left": null,
      "right": null,
      "value": 7,
    },
    "right": Node {
      "left": null,
      "right": null,
      "value": 9,
    },
    "value": 8,
  },
  Node {
    "left": null,
    "right": Node {
      "left": Node {
        "left": null,
        "right": null,
        "value": 16,
      },
      "right": null,
      "value": 17,
    },
    "value": 12,
  },
  Node {
    "left": null,
    "right": null,
    "value": 7,
  },
  Node {
    "left": null,
    "right": null,
    "value": 9,
  },
  Node {
    "left": Node {
      "left": null,
      "right": null,
      "value": 16,
    },
    "right": null,
    "value": 17,
  },
  Node {
    "left": null,
    "right": null,
    "value": 16,
  },
]
`;

exports[`BinarySearchTree #dfsPre should traverse then tree in preorder 1`] = `
Array [
  Node {
    "left": Node {
      "left": Node {
        "left": null,
        "right": null,
        "value": 7,
      },
      "right": Node {
        "left": null,
        "right": null,
        "value": 9,
      },
      "value": 8,
    },
    "right": Node {
      "left": null,
      "right": Node {
        "left": Node {
          "left": null,
          "right": null,
          "value": 16,
        },
        "right": null,
        "value": 17,
      },
      "value": 12,
    },
    "value": 10,
  },
  Node {
    "left": Node {
      "left": null,
      "right": null,
      "value": 7,
    },
    "right": Node {
      "left": null,
      "right": null,
      "value": 9,
    },
    "value": 8,
  },
  Node {
    "left": null,
    "right": null,
    "value": 7,
  },
  Node {
    "left": null,
    "right": null,
    "value": 9,
  },
  Node {
    "left": null,
    "right": Node {
      "left": Node {
        "left": null,
        "right": null,
        "value": 16,
      },
      "right": null,
      "value": 17,
    },
    "value": 12,
  },
  Node {
    "left": Node {
      "left": null,
      "right": null,
      "value": 16,
    },
    "right": null,
    "value": 17,
  },
  Node {
    "left": null,
    "right": null,
    "value": 16,
  },
]
`;

exports[`BinarySearchTree #insert should insert a node when 0 nodes 1`] = `
BinarySearchTree {
  "root": Node {
    "left": null,
    "right": null,
    "value": 1,
  },
}
`;

exports[`BinarySearchTree #insert should insert a node when multiple nodes 1`] = `
BinarySearchTree {
  "root": Node {
    "left": Node {
      "left": Node {
        "left": null,
        "right": null,
        "value": 7,
      },
      "right": Node {
        "left": null,
        "right": null,
        "value": 9,
      },
      "value": 8,
    },
    "right": Node {
      "left": null,
      "right": Node {
        "left": null,
        "right": null,
        "value": 17,
      },
      "value": 12,
    },
    "value": 10,
  },
}
`;
